driver-mapping/src/main/java/com/datastax/driver/mapping/QueryType.java

```diff
@@ -56,84 +60,100 @@ public static QueryType slice(int startBoundSize, boolean startInclusive, int en
         return new QueryType(reversed ? Kind.REVERSED_SLICE : Kind.SLICE, startBoundSize, startInclusive, endBoundSize, endInclusive);
     }
 
-    String makePreparedQueryString(TableMetadata table, EntityMapper<?> mapper) {
+    String makePreparedQueryString(TableMetadata table, EntityMapper<?> mapper, Mapper.Option... options) {
         switch (kind) {
-            case SAVE:
-                {
-                    Insert insert = table == null
-                                  ? insertInto(mapper.getKeyspace(), mapper.getTable())
-                                  : insertInto(table);
-                    for (ColumnMapper<?> cm : mapper.allColumns())
-                        insert.value(cm.getColumnName(), bindMarker());
-                    return insert.toString();
-                }
-            case GET:
-                {
-                    Select select = table == null
-                                  ? select().all().from(mapper.getKeyspace(), mapper.getTable())
-                                  : select().all().from(table);
-                    Select.Where where = select.where();
-                    for (int i = 0; i < mapper.primaryKeySize(); i++)
-                        where.and(eq(mapper.getPrimaryKeyColumn(i).getColumnName(), bindMarker()));
-                    return select.toString();
+            case SAVE: {
+                Insert insert = table == null
+                    ? insertInto(mapper.getKeyspace(), mapper.getTable())
+                    : insertInto(table);
+                for (ColumnMapper<?> cm : mapper.allColumns())
+                    insert.value(cm.getColumnName(), bindMarker());
+                if (options != null) {
+                    Insert.Options usings = insert.using();
+                    for (Mapper.Option opt : options) {
+                        if (opt instanceof Mapper.Option.Ttl)
+                            // Use Bind Marker here so we don't re-prepare statements for different options values
+                            usings.and(ttl(bindMarker()));
+                        else
+                            usings.and(timestamp(bindMarker()));
```

Comment:
We could make this a bit prettier by turning it into an abstract method on `Mapper.Option`, for example:
```java
for (Mapper.Option opt : options)
    opt.appendTo(usings);
```
Same thing when we bind the values in `Mapper`.
To handle the fact that options don't apply to all queries, we could have a method `boolean Option.isValidFor(QueryType)`.